iwyu.cc

```diff
@@ -618,8 +618,8 @@ class BaseAstVisitor : public RecursiveASTVisitor<Derived> {
     clang::Sema& sema = compiler_->getSema();
     DeclContext::lookup_result ctors = sema.LookupConstructors(decl);
     for (NamedDecl* ctor_lookup : ctors) {
-      // Ignore templated constructors.
-      if (isa<FunctionTemplateDecl>(ctor_lookup))
+      // Ignore templated or inheriting constructors.
+      if (!isa<CXXConstructorDecl>(ctor_lookup))
```

Comment:
I think it would be nicer to do this as a `dyn_cast` now:

     CXXConstructorDecl* ctor = dyn_cast<CXXConstructorDecl>(ctor_lookup);

     // Ignore templated or inheriting constructors
     if (!ctor)
       continue;

    ...

Reply:
Hmm, but it also makes this code a little more bug-prone. The current code explicitly spells out FunctionTemplateDecl as an exception to the norm (CXXConstructorDecl). It would be more conservative and informative to just add UsingDecl (which I guess this is?), something like:

    // Ignore templated and inheriting constructors.
    if (isa<FunctionTemplateDecl>(ctor_lookup) || isa<UsingDecl>(ctor_lookup))
      continue;

and then leave the `cast` as-is, to make sure we break if something unexpected creeps in here.